File: src/apps/Andriod/app/src/main/java/com/dkellycollins/triggerfinger/entity/impl/Player.java
package com.dkellycollins.triggerfinger.entity.impl;

import com.dkellycollins.triggerfinger.entity.IPlayer;

public class Player implements IPlayer {

    private final int _id;

    private int _collidableId;

    public Player(int id, int collidableId) {
        _id = id;
        _collidableId = collidableId;
    }

    @Override
    public int getCollidableId() {
        return _collidableId;
    }

    @Override
    public int getId() {
        return _id;
    }

    public void setCollidableId(int collidableId) {
        _collidableId = collidableId;
    }
}

File: src/apps/Android-Kotlin/app/src/main/kotlin/com/dkellycollins/triggerfinger/data/entity/impl/Player.kt
package com.dkellycollins.triggerfinger.data.entity.impl

import com.dkellycollins.triggerfinger.data.entity.IPlayer

import java.io.Serializable

class Player(override val id: Int, override val collidableId: Int, override val invincibleTimerId: Int, override var weaponId: Int, override var score: Int, health: Byte) : IPlayer, Serializable {
    override var health: Byte = 0
        get() = 0

    init {
        this.health = health
    }
}
